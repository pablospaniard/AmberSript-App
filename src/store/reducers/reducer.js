import * as constants from '../constants'

const initialStore = {
  err: null,
  loading: null,
  data: []
}

const dataReducer = (state = initialStore, action) => {
  switch (action.type) {
    case constants.DATA_FETCH_START:
      return { ...state, loading: true }
    case constants.DATA_FETCH_SUCCESS:
      return {
        ...state,
        loading: false,
        data: action.payload.data,
        src: action.payload.src
      }
    case constants.DATA_FETCH_FAIL:
      return { ...state, loading: false, err: action.payload }
    default:
      return state
  }
}

export default dataReducer
