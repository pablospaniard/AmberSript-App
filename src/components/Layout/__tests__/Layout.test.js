import React from 'react'
import { createStore } from 'redux'
import { Provider } from 'react-redux'
import { render, cleanup } from 'react-testing-library'

import reducer from '../../../store/reducers/reducer'
import { Layout } from '..'

describe('Layout should works properly', () => {
  function renderWithRedux(
    ui,
    { initialState, store = createStore(reducer, initialState) } = {}
  ) {
    return {
      ...render(<Provider store={store}>{ui}</Provider>),
      store
    }
  }

  const fetchData = jest.fn()

  afterEach(cleanup)

  it('Should match snapshot', () => {
    const { container } = renderWithRedux(<Layout fetchData={fetchData} />)
    expect(container.firstChild).toMatchSnapshot('Layout_snapshot_1')
  })
})
