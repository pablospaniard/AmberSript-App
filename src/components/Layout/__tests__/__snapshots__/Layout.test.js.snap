// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout should works properly Should match snapshot: Layout_snapshot_1 1`] = `
<div
  class="MuiGrid-container-1 Layout"
>
  <div
    class="MuiGrid-item-2 MuiGrid-grid-xs-12-41"
  >
    <div
      class="MuiGrid-container-1 Header"
    >
      <div
        class="MuiGrid-item-2 MuiGrid-grid-xs-2-31"
      >
        Logo
      </div>
      <div
        class="MuiGrid-item-2 MuiGrid-grid-xs-8-37"
      >
        Your Ambers
      </div>
      <div
        class="MuiGrid-item-2 MuiGrid-grid-xs-2-31"
      >
        Account
      </div>
    </div>
  </div>
  <div
    class="MuiGrid-item-2 MuiGrid-grid-xs-12-41"
  >
    <div>
      Content
    </div>
  </div>
</div>
`;
